### PR TITLE
UCT/CUDA: drain cudaLastError if it failed to get cudaPointerGetAttri…

### PR DESCRIPTION
**Pull request by bureddy**
…butes

### CODE REVIEW

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/6397/ for details (Mellanox internal link).

**Comment by bureddy** [comment]
bot:mlx:retest

**Comment by mellanox-github** [comment]
Test PASSed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/6398/ for details (Mellanox internal link).

File: src/uct/cuda/gdr_copy/gdr_copy_md.c [hunk 1]
```diff
@@ -239,6 +239,10 @@ static int uct_is_gdr_copy_mem_type_owned(uct_md_h md, void *addr, size_t length
     } else if (memory_type == CU_MEMORYTYPE_DEVICE) {
         return 1;
     }
+
+    /*reset cuda error if it is failed to get pointer attributes*/
+    cudaGetLastError();
```
Comment (yosefe) [inline src/uct/cuda/gdr_copy/gdr_copy_md.c]: @bureddy pls unite these to one function, maybe somewhere in cuda/base. it seems we have to every bug in multiple places..

**Comment by yosefe** [comment]
bot:bgate:retest

**Pull request merged**